### PR TITLE
fix: platform upgrade and ui sharing-dialog text fixes [v37]

### PR DESCRIPTION
**Pull request by jenniferarnesen**
Fixes in this PR:

* Clear the cache between users and on log out to prevent sensitive data being shared (https://jira.dhis2.org/browse/TECH-682)
* New update prompt (https://jira.dhis2.org/browse/LIBS-231)
* Enable translations for sharing dialog  (for screenshots, check https://github.com/dhis2/ui/pull/768)
* styled-jsx upgrade was a peer dependency requirement
* remove External users from sharing-dialog:  
image

### CODE REVIEW

**Comment by ghost** [comment]
I wasn't directly involved with the service worker updates to the app-runtime, @KaiVandivier will know about that than I do. I see that app-runtime has not yet been updated to the version that includes client-side caching. If you want that you'll want v8 of the app-runtime. See the changes here: https://github.com/dhis2/sms-configuration-app/pull/300/files (you'll also need a matching ui and cli-app-scripts), let me know if you have any questions or if I can help out!

**Comment by KaiVandivier** [comment]
Hey @ismay 🙂  Since the app-runtime major version bump happened after the code freeze, I figured I would continue back-porting PWA changes to the `7.x`-type maintenance branches so that apps didn't _need_ to use any new breaking changes before the 2.37 release, but then I'd leave it up to @dhis2/team-analytics to decide which version they want to go with for now

**Comment by ghost** [comment]
> Hey @ismay 🙂 Since the app-runtime major version bump happened after the code freeze, I figured I would continue back-porting PWA changes to the `7.x`-type maintenance branches so that apps didn't _need_ to use any new breaking changes before the 2.37 release, but then I'd leave it up to @dhis2/team-analytics to decide which version they want to go with for now

Sounds good. So then I guess you're probably the one to ask for specific feedback on this app-runtime update. I'll move my assignment to you if you don't mind!

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
         "redux-logger": "^3.0.6",
         "redux-thunk": "^2.3.0",
         "reselect": "^4.0.0",
-        "styled-jsx": "3.3.2"
+        "styled-jsx": "^4"
```
Comment (KaiVandivier) [inline package.json]: Were there any issues with styles before this version bump? Just curious about the effects of a possible version mismatch

**Review by KaiVandivier — APPROVED** [review]
LGTM!

**Comment by jenniferarnesen** [comment]
@ismay Sorry, I should have clearly stated why I added you to the PR. It was because of the `styled-jsx` bump. I knew there have previously been problems with version mismatches, so I wanted to check with you on this.

**Pull request merged**